tests/integration_test/ota_pal/aws_test_ota_pal.c

```diff
@@ -0,0 +1,1137 @@
+    xOtaFile.fileSize = sizeof( ucDummyData ) * testotapalNUM_WRITE_BLOCKS;
+    /* TEST: Write many bytes of data. */
+
+    xOtaFile.pFilePath = ( uint8_t * ) otatestpalFIRMWARE_FILE;
+    xOtaStatus = otaPal_CreateFileForRx( &xOtaFile );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, OTA_PAL_MAIN_ERR( xOtaStatus ) );
+
+    if( TEST_PROTECT() )
+    {
+        int lIndex = 0;
+
+        for( lIndex = 0; lIndex < testotapalNUM_WRITE_BLOCKS; lIndex++ )
+        {
+            blocksWritten = otaPal_WriteBlock( &xOtaFile, lIndex * sizeof( ucDummyData ), ucDummyData, sizeof( ucDummyData ) );
+            TEST_ASSERT_EQUAL_INT( sizeof( ucDummyData ), blocksWritten );
+        }
+
+        /* Sufficient delay for flash write to complete. */
+        vTaskDelay( pdMS_TO_TICKS( testotapalWRITE_BLOCKS_DELAY_MS ) );
+    }
+}
+
+/**
+ * Call otaPal_ActivateNewImage() and verify success. This function is expected to
+ * reset the device, so this test is only supported on the Windows Simulator environment.
+ * The Windows Simulator environment does not reset.
+ */
+TEST( Full_OTA_PAL, otaPal_ActivateNewImage_HappyPath )
+{
+    #ifdef WIN32
+        OtaPalStatus_t xOtaStatus = otaPal_ActivateNewImage( &xOtaFile );
+
+        TEST_ASSERT_EQUAL_INT( OtaPalSuccess, OTA_PAL_MAIN_ERR( xOtaStatus ) );
+    #endif
+}
+
+/**
+ * @brief Set the platform state to self-test and verify success.
+ */
+TEST( Full_OTA_PAL, otaPal_SetPlatformImageState_SelfTestImageState )
+{
+    OtaPalStatus_t xOtaStatus;
+    OtaImageState_t eImageState = OtaImageStateUnknown;
+    int16_t blocksWritten;
+
+    /* Create a local file again using the PAL. */
+    xOtaFile.pFilePath = ( uint8_t * ) otatestpalFIRMWARE_FILE;
+    xOtaFile.fileSize = sizeof( ucDummyData );
+
+    xOtaStatus = otaPal_CreateFileForRx( &xOtaFile );
+    TEST_ASSERT_EQUAL( OtaPalSuccess, OTA_PAL_MAIN_ERR( xOtaStatus ) );
+
+    /* We still want to close the file if the test fails. */
+    if( TEST_PROTECT() )
+    {
+        /* Write data to the file. */
+        blocksWritten = otaPal_WriteBlock( &xOtaFile,
+                                           0,
+                                           ucDummyData,
+                                           sizeof( ucDummyData ) );
+        TEST_ASSERT_EQUAL( sizeof( ucDummyData ), blocksWritten );
+
+        /* Set the image state. */
+        eImageState = OtaImageStateTesting;
+        xOtaStatus = otaPal_SetPlatformImageState( &xOtaFile, eImageState );
+        TEST_ASSERT_EQUAL_INT( OtaPalSuccess, OTA_PAL_MAIN_ERR( xOtaStatus ) );
+
+        /* Verify that image state was saved correctly. */
+
+        /* [**]All platforms need a reboot of a successfully close image in order to return
+         * OtaPalImageStatePendingCommit from otaPal_GetPlatformImageState(). So this cannot be tested.
+         */
+
+        /*ePalImageState = otaPal_GetPlatformImageState();
+         * TEST_ASSERT_EQUAL_INT( OtaPalImageStatePendingCommit, ePalImageState ); */
```

Comment:
I think this was originally left there to show the test that we are ignoring because of device limitations. But I agree this could be confusing so I have removed this. There is still a comment above here explaining why we aren't doing that specific check.